plugins/Login/Security/BruteForceDetection.php

```diff
@@ -144,4 +151,93 @@ private function getDateTimeSubMinutes($minutes)
     {
         return $this->getNow()->subPeriod($minutes, 'minute')->getDatetime();
     }
+
+    // TODO: need to make sure bruteforcedetection happens after any ip allowlists/blocklists have been used
+    public function isUserLoginBlocked($login)
+    {
+        $count = $this->getTotalLoginAttemptsInLastHourForLogin($login);
+        if (!$this->hasTooManyTriesOverallInlastHour($count)) {
+            return false;
+        }
+
+        if (!$this->hasNotifiedUserAboutSuspiciousLogins($login)) {
+            $this->sendSuspiciousLoginsEmailToUser($login, $count);
+        }
+
+        return true;
+    }
+
+    private function getTotalLoginAttemptsInLastHourForLogin($login)
```

Comment:
be good to move some of the private methods maybe in some other class and then have them public so they can be tested

Reply:
Moved to a new Model class. Couldn't think of a class that wouldn't require the calling method isUserLoginBlocked, so created a new Model class for Login. Integration tests were added too.